scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,792 @@
+                video: false,
+                thumbnail: false,
+            },
+            loadingText: '',
+            scale: 1,
+            prevVideoHeight: 1,
+            thumbnails: [],
+            videoSrc: '',
+            article: this.props.article,
+        };
+        this.videoWrapper = null;
+        this.videoContainerSize = 0;
+        this.videoToolsSize = 0;
+        this.hasTransitionRun = true;
+        this.reactCropInitImage = '';
+
+        this.handleClose = this.handleClose.bind(this);
+        this.handleTrim = this.handleTrim.bind(this);
+        this.handleRotate = this.handleRotate.bind(this);
+        this.handleRotateTransitionEnd = this.handleRotateTransitionEnd.bind(this);
+        this.handleCrop = this.handleCrop.bind(this);
+        this.handleToggleLoading = this.handleToggleLoading.bind(this);
+        this.handleToggleCrop = this.handleToggleCrop.bind(this);
+        this.playPause = this.playPause.bind(this);
+        this.handleQualityChange = this.handleQualityChange.bind(this);
+        this.handleSave = this.handleSave.bind(this);
+        this.getCropRotate = this.getCropRotate.bind(this);
+        this.setVideoWrapper = this.setVideoWrapper.bind(this);
+        this.setVideoToolsSize = this.setVideoToolsSize.bind(this);
+        this.setVideoContainerSize = this.setVideoContainerSize.bind(this);
+        this.showErrorMessage = this.showErrorMessage.bind(this);
+        this.checkIsDirty = this.checkIsDirty.bind(this);
+    }
+
+    componentDidMount() {
+        const canvas = document.createElement('canvas');
+        const ctx = canvas.getContext('2d');
+
+        // initilize image for React Crop
+        // this is the boudary of crop area, user can't move or draw crop outside of this so image need
+        // to be big enough to cover entire video even once rotated
+        canvas.width = 2000;
+        canvas.height = 2000;
+        if (ctx == null) {
+            return;
+        }
+
+        ctx.globalAlpha = 0;
+        ctx.fillStyle = 'rgba(0, 0, 200, 0.5)';
+        const {gettext} = this.props.superdesk.localization;
+
+        this.handleToggleLoading(true, gettext('Loading video...'));
+        this.handleCheckingVideo(false, () => {
+            const text = gettext('Video is editing, please wait...');
+
+            if (this.state.loadingText !== text) {
+                this.setState({
+                    loadingText: text,
+                });
+            }
+        });
+        this.reactCropInitImage = canvas.toDataURL();
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.intervalThumbnails);
+        clearInterval(this.intervalCheckVideo);
+    }
+
+    handleClose() {
+        this.props.onClose();
+
+        this.props.superdesk.entities.article.patch(
+            this.state.article,
+            {
+                renditions: this.state.article.renditions,
+            },
+            {patchDirectlyAndOverwriteAuthoringValues: true},
```

Comment:
@tomaskikutis We just noticed that this flag will make updating video thumbnail on _article editor view_ no longer work.
Steps to reproduce:
- Open video in edit mode
- Open Video editor then close it
- Change Video thumbnail and click save

You could see in the `devtools` that although we clicked `save`, no `PATCH` request is made. Logging `AuthoringService` showed that `originalItem` and `item` are the same.

Reply:
Something is wrong with my local video server and I can't start it. Oleg is now working on deploying a test instance with video server enabled and I will take a look at this when the test instance is ready.

Reply:
I just found that there's an issue with `renditions.thumbnail` - you should replace it with `renditions.viewImage` in all places. `thumbnail` rendition is only for backwards-compatibility and will likely be dropped in the future.

> You could see in the devtools that although we clicked save, no PATCH request is made.

Could you record a video with the issue? It might be that because of a large image - if it hasn't finished uploading there will be nothing to save.

Here are a few videos from my testing. There are a few UI glitches when rotating & capturing thumbnails with non-standard sizes. Looks like timeline is missing too.

https://drive.google.com/file/d/1kaQHGAUjbGmg5GdjcuuM4jmr33qX31o2/view?usp=sharing
https://drive.google.com/file/d/1-HMSp2VJHBD7NYozi6GKKYm60TNKrUiO/view?usp=sharing

Reply:
> https://drive.google.com/file/d/1kaQHGAUjbGmg5GdjcuuM4jmr33qX31o2/view?usp=sharing

It seems style is broken, could you check if css is generated and loaded correctly? It should be [like this](https://drive.google.com/file/d/1CnfBdl6kQAaUL3p1e1guwh9XWGj1u9Ub/view?usp=sharing) (1100x500).

> Looks like timeline is missing too

could you send me server error response so I can check it? Just check which request has error with `?action=timeline` in its url is fine.

Regarding thumbnail, after open video editor view and exit, thumbnail is uploaded successfully but save does not apply, you could see on 40s, there should be a PATCH request upon clicking save, here is video: https://drive.google.com/file/d/1vph3ulhmoqXdZA9_wc4-dcIL8m7-heik/view?usp=sharing